File: lib_arch/src/main/java/com/yyxnb/arch/delegate/IActivityDelegate.java
package com.yyxnb.arch.delegate;

import android.app.Activity;
import android.arch.lifecycle.DefaultLifecycleObserver;
import android.arch.lifecycle.Lifecycle;
import android.arch.lifecycle.OnLifecycleEvent;
import android.os.Bundle;

/**
 * @author yyx
 */
public interface IActivityDelegate extends DefaultLifecycleObserver {

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    void onCreate(Bundle savedInstanceState);

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    void onStart();

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    void onResume();

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    void onPause();

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    void onStop();

    void onSaveInstanceState(Activity activity, Bundle outState);

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    void onDestroy();

}

File: lib_arch/src/main/java/com/yyxnb/arch/delegate/IActivityDelegate.kt
package com.yyxnb.arch.delegate

import android.app.Activity
import android.os.Bundle
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.OnLifecycleEvent
import com.yyxnb.common.interfaces.ILifecycle

/**
 * @author yyx
 */
interface IActivityDelegate : ILifecycle {

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun onCreate(savedInstanceState: Bundle?)

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    fun onStart()

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    fun onResume()

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    fun onPause()

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    fun onStop()

    fun onSaveInstanceState(activity: Activity?, outState: Bundle?)

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    fun onDestroy()
}